src/abstractdataframe/iteration.jl

```diff
@@ -28,16 +37,52 @@ Base.length(itr::DFRowIterator) = size(itr.df, 1)
 Base.getindex(itr::DFRowIterator, i) = DataFrameRow(itr.df, i)
 
 # Iteration by columns
+"""
+    DFColumnIterator{<:AbstractDataFrame}
+
+Iterator over columns of an `AbstractDataFrame`.
+Each returned value is a tuple consisting of column name and column vector.
+
+A value of this type is returned by the [`eachcol`](@link) function.
+"""
 struct DFColumnIterator{T <: AbstractDataFrame}
     df::T
 end
+
+"""
+    eachcol(df::AbstractDataFrame)
+
+Return a `DFColumnIterator` that iterates an `AbstractDataFrame` column by column.
+Iteration returns a tuple consisting of column name and column vector.
+
+`DFColumnIterator` has a custom implementation of the `map` function which
+returns a `DataFrame` and assumes that a function argument passed do
+the `map` function accepts takes only a column vector.
+
+**Examples**
+
+```jldoctest
+julia> df = DataFrame(x=1:4, y=11:14)
+4×2 DataFrame
+│ Row │ x     │ y     │
+│     │ Int64 │ Int64 │
+├─────┼───────┼───────┤
+│ 1   │ 1     │ 11    │
+│ 2   │ 2     │ 12    │
+│ 3   │ 3     │ 13    │
+│ 4   │ 4     │ 14    │
+
+julia> map(sum, eachcol(df)) == DataFrame(x=10, y=50)
```

Comment:
I mean that you can just print the output (it's weird to check for equality).

Reply:
ok